csharp/RowOriented/ParquetRowReader.cs

```diff
@@ -13,29 +13,30 @@ public sealed class ParquetRowReader<TTuple> : IDisposable
     {
         internal delegate void ReadAction(ParquetRowReader<TTuple> parquetRowReader, TTuple[] rows, int length);
 
-        internal ParquetRowReader(string path, ReadAction readAction, MappedField[] fields)
-            : this(new ParquetFileReader(path), readAction, fields)
+        internal ParquetRowReader(string path, ReadAction readAction, MappedField[] fields, LogicalTypeFactory? logicalTypeFactory = null, LogicalReadConverterFactory? logicalReadConverterFactory = null)
```

Comment:
I think the issue might be that the `logicalTypeFactory` isn't being passed through to the underlying `ParquetFileReader`. The constructors here all ignore that parameter.

There's a comment in the docs that matches with the behaviour you describe: https://github.com/G-Research/ParquetSharp/blob/7d8b7bc995f072ab11e829819368fa5806289080/docs/TypeFactories.md?plain=1#L15

Reply:
I'm a bit confused about this. I noticed that for the writing scenario, the `ParquetRowWriter` constructors all admit the `LogicalTypeFactory` as a parameter to the `ParquetFileWriter`, therefore the common constructor only requires a `LogicalWriteConverterFactory` to work. On the other hand, the `ParquetFileReader` doesn't admit a `LogicalTypeFactory`. Does this mean I need to set both in the common constructor for reading?

### ParquetRowWriter.cs
![image](https://github.com/G-Research/ParquetSharp/assets/97265671/95f28b75-0653-4482-924a-fd6326bba0a5)

### ParquetRowReader.cs
![image](https://github.com/G-Research/ParquetSharp/assets/97265671/96879217-dea7-4b48-85de-4a230ef092eb)

Reply:
I think this is because the writer requires the logical type factory to be present at construction time to convert from the `Column` types to the schema to write, but the reader only needs it when you start reading data, so doesn't take it in the constructor. But you will still need to set both for the mapping to work to correctly.